hasher-matcher-actioner/src/OpenMediaMatch/blueprints/matching.py

```diff
@@ -247,27 +252,39 @@ def lookup_post():
 
 def lookup(signal, signal_type_name):
     current_app.logger.debug("performing lookup")
-    raw_results = lookup_signal(signal, signal_type_name)
+    results_by_bank_content_id = {
+        r.metadata: r for r in query_index(signal, signal_type_name)
+    }
     storage = get_storage()
-    current_app.logger.debug("getting bank content")
-    current_app.logger.debug(raw_results)
-    contents = storage.bank_content_get(raw_results)
-    enabled = [c for c in contents if c.enabled]
+    contents = storage.bank_content_get(results_by_bank_content_id)
+    enabled_content = [c for c in contents if c.enabled]
     current_app.logger.debug(
-        "lookup matches %d content ids (%d enabled)", len(contents), len(enabled)
+        "lookup matches %d content ids (%d enabled_content)",
+        len(contents),
+        len(enabled_content),
     )
-    if not enabled:
-        return []
-    banks = {c.bank.name: c.bank for c in enabled}
+    banks = {c.bank.name: c.bank for c in enabled_content}
     rand = random.Random(request.args.get("seed"))
     coinflip = rand.random()
-    enabled_banks = [
+    enabled_banks = {
         b.name for b in banks.values() if b.matching_enabled_ratio >= coinflip
-    ]
+    }
     current_app.logger.debug(
-        "lookup matches %d banks (%d enabled)", len(banks), len(enabled_banks)
+        "lookup matches %d banks (%d enabled_banks)", len(banks), len(enabled_banks)
     )
-    return enabled_banks
+    results = defaultdict(list)
+    for content in enabled_content:
+        if content.bank.name not in enabled_banks:
+            continue
+
+        match = results_by_bank_content_id.get(content.id)
+        results[content.bank.name].append(
+            {
+                "content_id": content.id,
```

Comment:
blocking: I missed this version of it, but I forsee the need for a "content id" down the line, and so we should make this clear it's the `bank_content_id`